generator/src/main/java/com/scottlogic/deg/generator/decisiontree/MaxStringLengthInjectingDecisionTreeFactory.java

```diff
@@ -0,0 +1,64 @@
+package com.scottlogic.deg.generator.decisiontree;
+
+import com.google.inject.Inject;
+import com.scottlogic.deg.generator.Profile;
+import com.scottlogic.deg.generator.constraints.atomic.IsStringShorterThanConstraint;
+import com.scottlogic.deg.generator.generation.GenerationConfig;
+import com.scottlogic.deg.generator.inputs.RuleInformation;
+import com.scottlogic.deg.schemas.v0_1.RuleDTO;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Decorator over a DecisionTreeFactory to inject a &lt;shorterThan X&gt; constraint at the root node for every field
+ */
+public class MaxStringLengthInjectingDecisionTreeFactory implements DecisionTreeFactory{
```

Comment:
Could we name this as a decorator if it is a decorator.

Reply:
I'd prefer not to include the name of the pattern being adopted in the name of the type.
`MaxStringLengthInjectingDecisionTreeFactoryDecorator` doesn't feel right.

It is a DecisionTreeFactory that achieves the creation of a factory by delegating some control to another type (by decorating it) - that's not relevant to any type that wants to use it.

The name of this type could be improved to something like `DefaultInjectingDecisionTreeFactory` , in the fullness of time, but it would currently imply something that it doesn't do.

Reply:
what about StringLengthTreeFactoryDecorator?

Reply:
Relatively long response to this thread:

`MaxStringLengthInjector` is far better than `XXXFactoryDecorator`.

The fact that this type decorates another type is irrelevant to a consumer. That it is a factory is important, as it conveys the meaning of it being responsible for creating something.

The downside to `MaxStringLengthInjector` is it implies that it requires something that will have string-lengths injected into, which is true in part, but mostly inaccurate. It is a factory that will create something and include this detail in the result. I prefer the current name for the following reasons:

- It is responsible for creating things, therefore should have `Factory` in its name
- It uses decoration to achieve the result, but this is irrelevant to consumers, so shouldn't include `Decorator` in the name
- It is responsible for 'injecting' max-string-lengths in the output, so this should be included in its name
- It is not responsible for 'injecting' anything else, it is not general purpose, so should not contain generalisations in its name.